### PR TITLE
Improve "ExecCtx::RunList()" with O(1) implementation.

### PR DESCRIPTION
**Pull request by LeiW000**
We don't have to iterate through the list for combining the two lists. They can be combined directly. It can improve the performance.

### CODE REVIEW

**Comment by LeiW000** [comment]
Is there anyone, who can help to review the PR?

**Comment by LeiW000** [comment]
Is there anything else I need to do?

**Comment by LeiW000** [comment]
@yashykt , can the PR be merged?

**Comment by LeiW000** [comment]
Can anyone help to take look at the PR?

**Comment by yashykt** [comment]
Sorry for the delay! I will take a look at this PR this week.
@ctiller would appreciate another pair of eyes here.

File: src/core/lib/iomgr/exec_ctx.cc [hunk 1]
```diff
@@ -106,27 +106,18 @@ void ExecCtx::Run(const DebugLocation& location, grpc_closure* closure,
...
+
+  if (list->head != nullptr) {
+    if (destlist->head == nullptr) {
+      destlist->head = list->head;
+      destlist->tail = list->tail;
+    } else {
+      destlist->tail->next_data.next = list->head;
+      destlist->tail = list->tail;
     }
-    c->scheduled = true;
```
Comment (yashykt) [inline src/core/lib/iomgr/exec_ctx.cc:121]: I think the approach probably works but you you are no longer setting `scheduled` any more

Comment (yashykt) [reply src/core/lib/iomgr/exec_ctx.cc:121]: nor are the other fields being set

Comment (LeiW000) [reply src/core/lib/iomgr/exec_ctx.cc:121]: The fields are defined in "#ifndef NDEBUG" for debugging. I'm wondering if we can ignore them here.

**Comment by LeiW000** [comment]
@yashykt @ctiller , thanks for your reviews. I decide to close the PR as I can't find a good way to make the change while keeping the debug code.

**Pull request closed without merging**